test/e2e/nodenetworkstate_test.go

```diff
@@ -52,17 +52,33 @@ var _ = Describe("NodeNetworkState", func() {
 			})
 			It("should have the linux bridge at currentState", func() {
 				for _, node := range nodes {
-					Eventually(func() []string {
-						var currentStateYaml nmstatev1.State
-						currentState(namespace, node, &currentStateYaml).ShouldNot(BeEmpty())
-
-						interfaces := interfaces(currentStateYaml)
-						Expect(interfaces).ToNot(BeEmpty(), "Node %s should have network interfaces", node)
+					interfacesForNode(node).Should(ContainElement("br1"))
+				}
+			})
+		})
+		Context("with a linux bridge absent", func() {
+			var (
+				br1Absent = nmstatev1.State(`interfaces:
+  - name: br1
+    type: linux-bridge
+    state: absent
+`)
+			)
 
-						return interfacesName(interfaces)
-					}, ReadTimeout, ReadInterval).Should(ContainElement("br1"))
+			BeforeEach(func() {
+				createBridgeAtNodes("br1")
+				updateDesiredState(namespace, br1Absent)
+			})
+			AfterEach(func() {
+				// If not br1 is going to be removed if created manually
+				resetDesiredStateForNodes(namespace)
+			})
+			It("should have the linux bridge at currentState", func() {
```

Comment:
wrong description? If linux bridge absent it should remove the linux bridge from host. Overall, I'd section it little diferently (just an idea for another PR).

```
Context("when there is a linux bridge configured on a node")
	// before each, create the bridge, after each, cleanup
	Context("and it is targeted as absent")
		// before each set desiredState
		It("should be removed from the host and reported state")
```

Reply:
I think you are checking old stuff

Reply:
Tests should be most of all easy to read. That I think means [Descriptions and organization of Contexts] > [Test Code Workflow] > [Test Code].